### ISSUE TITLE
Support using third-party Java agents for Gradle daemon process with configuration cache enabled

### ISSUE THREAD

**Issue by mlopatkin** [gradle/gradle#25979]
### Expected Behavior

A user can apply java agent to the Gradle daemon process, and build with configuration cache enabled. 

### Current Behavior (optional)

* Attached agent is reported as a configuration cache problem when using TestKit to run the build (see also gradle/gradle#25929), even if the agent does not register instrumentation.
* Legacy instrumentation mode exposes instrumented bytecode to third-party agents (which is problematic for e.g. Jacoco's online instrumentation).

### Context

TestKit is the primary concern, as plugin authors want to test their plugins with the configuration cache. 

FWIW, Jacoco is not fully supported with TestKit yet (see #1465, #16603).

**Comment by autonomousapps** [comment]
This impacts usage of Gradle TestKit in the Cash backend. Close to 10M LOC and ~2000 Gradle modules. This restriction poses a material barrier to rolling out more thorough testing of our suite of Gradle plugins for managing this large set of builds.

**Comment by CRogers** [comment]
The Java debugger is a Java agent. I think this means it's not possible to debug testkit runs that use configuration cache without the gradle process failing? This is a pretty annoying tbh - it can mean the test fails early for no reason.

**Comment by CRogers** [comment]
This is becoming a real blocker to our huge configuration cache migration.

**Comment by autonomousapps** [comment]
I agree this is annoying, but I have this [workaround](https://github.com/autonomousapps/dependency-analysis-gradle-plugin/blob/main/src/functionalTest/groovy/com/autonomousapps/AbstractProject.groovy#L105-L108) that disables CC when debugging. Works for me, though I'd prefer to debug with CC as well 🤷
